From e2fd5b3870c4118e129bf6764d0c69e55530b273 Mon Sep 17 00:00:00 2001
From: Ethan Chen <ethan84@andestech.com>
Date: Fri, 1 Sep 2023 13:43:10 +0800
Subject: [PATCH 050/241] target/riscv: Fix typo

Signed-off-by: Ethan Chen <ethan84@andestech.com>
---
 target/riscv/cpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/target/riscv/cpu.c b/target/riscv/cpu.c
index 00b269cc8f..27045b9716 100644
--- a/target/riscv/cpu.c
+++ b/target/riscv/cpu.c
@@ -1190,7 +1190,7 @@ static void riscv_cpu_reset_hold(Object *obj)
         mcc->parent_phases.hold(obj);
     }
 #ifndef CONFIG_USER_ONLY
-    andes_spec_csr_reset_common(env);
+    andes_csr_reset_common(env);
     env->misa_mxl = env->misa_mxl_max;
     env->priv = PRV_M;
     env->mstatus &= ~(MSTATUS_MIE | MSTATUS_MPRV);
-- 
2.34.1

