From dace5936177d00cee7f498fc50142f19780740bb Mon Sep 17 00:00:00 2001
From: Ethan Chen <ethan84@andestech.com>
Date: Wed, 4 Oct 2023 17:43:46 +0800
Subject: [PATCH 095/241] target/riscv: pmp: Fix DLM is not checked when ILM is
 enabled.

Signed-off-by: Ethan Chen <ethan84@andestech.com>
---
 target/riscv/pmp.c | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/target/riscv/pmp.c b/target/riscv/pmp.c
index 814ddd06c6..bd842d3c5f 100644
--- a/target/riscv/pmp.c
+++ b/target/riscv/pmp.c
@@ -630,6 +630,10 @@ target_ulong pmp_get_tlb_size(CPURISCVState *env, target_ulong addr)
     target_ulong pmp_ea;
     target_ulong tlb_sa = addr & ~(TARGET_PAGE_SIZE - 1);
     target_ulong tlb_ea = tlb_sa + TARGET_PAGE_SIZE - 1;
+    target_ulong ilm_sa = env->ilm_base;
+    target_ulong ilm_ea = env->ilm_base + env->ilm_size - 1;
+    target_ulong dlm_sa = env->dlm_base;
+    target_ulong dlm_ea = env->dlm_base + env->dlm_size - 1;
     int i;
 
     /*
@@ -667,18 +671,19 @@ target_ulong pmp_get_tlb_size(CPURISCVState *env, target_ulong addr)
         }
     }
 
-    if (env->andes_csr.csrno[CSR_MILMB] & 0x1) {
-        if (env->ilm_base <= tlb_sa &&
-            env->ilm_base + env->ilm_size - 1 >= tlb_ea) {
+    if (env->andes_csr.csrno[CSR_MILMB] & 0x1 ) {
+        if (ilm_sa <= tlb_sa && ilm_ea >= tlb_ea) {
             return TARGET_PAGE_SIZE;
-        } else {
+        } else if ((ilm_sa >= tlb_sa && ilm_sa <= tlb_ea) ||
+                   (ilm_ea >= tlb_sa && ilm_ea <= tlb_ea)) {
             return 1;
         }
-    } else if (env->andes_csr.csrno[CSR_MDLMB] & 0x1) {
-        if (env->dlm_base <= tlb_sa &&
-            env->dlm_base + env->dlm_size - 1 >= tlb_ea) {
-            return  TARGET_PAGE_SIZE;
-        } else {
+    }
+    if (env->andes_csr.csrno[CSR_MDLMB] & 0x1 ) {
+        if (dlm_sa <= tlb_sa && dlm_ea >= tlb_ea) {
+            return TARGET_PAGE_SIZE;
+        } else if ((dlm_sa >= tlb_sa && dlm_sa <= tlb_ea) ||
+                   (dlm_ea >= tlb_sa && dlm_ea <= tlb_ea)) {
             return 1;
         }
     }
-- 
2.34.1

