From 70cdd998ca04e898dcba4af938c64be12273f6d7 Mon Sep 17 00:00:00 2001
From: Yu-Ming Chang <yumin686@andestech.com>
Date: Tue, 30 Apr 2024 16:50:28 +0800
Subject: [PATCH 231/241] target/riscv: Adjust write mask of mstatus for Andes
 cores.

Signed-off-by: Yu-Ming Chang <yumin686@andestech.com>
---
 target/riscv/csr.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/target/riscv/csr.c b/target/riscv/csr.c
index 9e6c7df3ee..97925af45c 100644
--- a/target/riscv/csr.c
+++ b/target/riscv/csr.c
@@ -1374,8 +1374,15 @@ static RISCVException write_mstatus(CPURISCVState *env, int csrno,
     if ((val ^ mstatus) & MSTATUS_MXR) {
         tlb_flush(env_cpu(env));
     }
-    mask = MSTATUS_SIE | MSTATUS_SPIE | MSTATUS_MIE | MSTATUS_MPIE |
-        MSTATUS_SPP | MSTATUS_MPRV | MSTATUS_SUM |
+
+    if (env_archcpu(env)->cfg.ext_XAndesV5Ops) {
+        if (riscv_has_ext(env, RVS)) {
+            mask |= MSTATUS_SIE | MSTATUS_SPIE | MSTATUS_SPP | MSTATUS_SUM;
+        }
+    } else {
+        mask |= MSTATUS_SIE | MSTATUS_SPIE | MSTATUS_SPP | MSTATUS_SUM;
+    }
+    mask |= MSTATUS_MIE | MSTATUS_MPIE | MSTATUS_MPRV |
         MSTATUS_MPP | MSTATUS_MXR | MSTATUS_TVM | MSTATUS_TSR |
         MSTATUS_TW | MSTATUS_VS;
 
-- 
2.34.1

